extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java

```diff
@@ -30,57 +34,74 @@
             .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
 
     private final SpringCloudConfigClientConfig springCloudConfigClientConfig;
-    private final String baseUri;
+    private final URI baseURI;
 
     public DefaultSpringCloudConfigClientGateway(SpringCloudConfigClientConfig springCloudConfigClientConfig) {
         this.springCloudConfigClientConfig = springCloudConfigClientConfig;
-        this.baseUri = determineBaseUri(springCloudConfigClientConfig);
+        try {
+            this.baseURI = determineBaseUri(springCloudConfigClientConfig);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
     }
 
-    private String determineBaseUri(SpringCloudConfigClientConfig springCloudConfigClientConfig) {
-        String baseUri = springCloudConfigClientConfig.url;
-        if (null == baseUri || baseUri.isEmpty()) {
-            throw new IllegalArgumentException("baseUri cannot be empty");
+    private URI determineBaseUri(SpringCloudConfigClientConfig springCloudConfigClientConfig) throws URISyntaxException {
+        String url = springCloudConfigClientConfig.url;
+        if (null == url || url.isEmpty()) {
+            throw new IllegalArgumentException("baseUrl cannot be empty");
```

Comment:
I would make the message more explicit by adding some context (typically which extension or the full property name)?

Reply:
Right, that makes sense